### PR TITLE
CTI API Helpers in expr

### PR DESCRIPTION
**Pull request by buixor**
- Add CTI API client in crowdsec that can be used from expr
 - Add `on_error` in profiles evaluation (and don't make error in expression lead to  500 from LAPI)

CTI helpers have various intended uses:

1. post-overflow whitelists

```yaml
name: crowdsecurity/cti-whitelist
description: "Whitelist known good IPs"
whitelist:
  reason: "Known FP"
  expression: 
    - "CrowdsecCTI(evt.Overflow.Alert.Source.IP).IsFalsePositive()"
```

2. custom profiles

```yaml
#if it's a known false-positive, cancel the decision and notify ourselves
name: known_fp
filters:
 - |
  Alert.Remediation == true && Alert.GetScope() == "Ip" &&
  CrowdsecCTI(Alert.GetValue()).IsFalsePositive()
notifications:
  - slack_known_fp_alert
on_error: ignore
on_success: break
---
#if IP has a good reputation, warn the user but carry on with the decision
name: warn_good_reputation
filters:
 - |
  Alert.Remediation == true && Alert.GetScope() == "Ip" &&
  CrowdsecCTI(Alert.GetValue()).GetMaliciousnessScore() < 0.3
notifications:
  - slack_potential_fp_alert
on_error: ignore
on_success: continue
---
name: default_ip_remediation
#debug: true
filters:
 - Alert.Remediation == true && Alert.GetScope() == "Ip"
decisions:
 - type: ban
   duration: 4h
on_success: break

```

3. Improved notification templates (ie. slack.yaml) :

```yaml
type: slack           # Don't change
name: slack_known_fp_alert   # Must match the registered plugin in the profile
log_level: info
format: |
  {{range . -}}
  {{$alert := . -}}
  :warning: *False Positive* :flag-{{$alert.Source.Cn}}: <https://app.crowdsec.net/cti/{{$alert.Source.IP}}|{{$alert.Source.IP}}> triggered *{{$alert.Scenario}}* ({{$alert.Source.AsName}}) 
  {{- $cti := $alert.Source.IP | CrowdsecCTI  -}}
  {{- range $fp := $cti.GetFalsePositives }} *{{$fp}}* {{ end -}}
  {{- end }}

webhook: https://hooks.slack.com/services/...
---
type: slack           # Don't change
name: slack_potential_fp_alert   # Must match the registered plugin in the profile
log_level: info
format: |
  
  {{range . -}}
  {{$alert := . -}}
  :flag-{{$alert.Source.Cn}}: <https://app.crowdsec.net/cti/{{$alert.Source.IP}}|{{$alert.Source.IP}}> triggered *{{$alert.Scenario}}* ({{$alert.Source.AsName}}) : Maliciousness Score is 
  {{- $cti := $alert.Source.IP | CrowdsecCTI  -}}
  {{" "}}{{mulf $cti.GetMaliciousnessScore 100 | floor}} %
  {{- end }}

webhook: https://hooks.slack.com/services/...
```

which looks like this:

image

or like that:

image

### CODE REVIEW

File: pkg/csconfig/api.go [hunk 1]
```diff
@@ -230,6 +258,14 @@ func (c *Config) LoadAPIServer() error {
 	if c.API.Server.OnlineClient == nil || c.API.Server.OnlineClient.Credentials == nil {
 		log.Printf("push and pull to Central API disabled")
 	}
+
+	log.Printf("Config load, c.API.CTI : %p", c.API.CTI)
```
Comment (LaurenceJJones) [inline pkg/csconfig/api.go:262]: Shouldn't the line be inside the nil catch as most people start will get nil at start?

File: pkg/csconfig/api.go [hunk 2]
```diff
@@ -44,6 +45,34 @@ type LocalApiClientCfg struct {
 	InsecureSkipVerify  *bool              `yaml:"insecure_skip_verify"` // check if api certificate is bad or not
 }
 
+type CTICfg struct {
+	Key          *string        `yaml:"key,omitempty"`
+	CacheTimeout *time.Duration `yaml:"cache_timeout,omitempty"`
+	CacheSize    *int           `yaml:"cache_size,omitempty"`
+	Enabled      bool           `yaml:"-"`
```
[the remark this answers is not in the document]
Comment (buixor) [reply pkg/csconfig/api.go:52]: I'm not sure to get the interest of this, let's chat about it :)

File: pkg/cticlient/cti_test.go [hunk 3]
```diff
@@ -0,0 +1,303 @@
+package cticlient
+
+// import (
```
[the remark this answers is not in the document]
Comment (buixor) [reply pkg/cticlient/cti_test.go:3]: leftover tests, blame @blotus ! (more seriously, I let this open so we decide what to do with it)

**Pull request merged**